### ISSUE TITLE
Required Fields nested inside ListBlock inside StructBlock inside StreamField missing asterisk

### ISSUE THREAD

**Issue by jslay-excella** [wagtail/wagtail#8047]
### Issue Summary
We have a page that uses StreamField, to include a custom StructBlock, which has a ListBlock, with a CharBlock. The CharBlock cannot be null, therefore required. However, there is no indication on the page that the CharBlock field cannot be empty.  [Issue on our side](https://github.com/cfpb/consumerfinance.gov/issues/3244)

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`
2. Edit home/models.py as follows...
```Python
from django.db import models

from wagtail.core.models import Page
from wagtail.core.fields import StreamField
from wagtail.core.blocks import CharBlock, ListBlock, StructBlock
from wagtail.admin.edit_handlers import StreamFieldPanel


class MyStructBlock(StructBlock):
    tags = ListBlock(CharBlock(required=True), help_text='CharBlocks missing asterisk')


class HomePage(Page):
    body = StreamField([
        ('thing', MyStructBlock())
    ])

    content_panels = Page.content_panels + [
        StreamFieldPanel('body'),
    ]
```
4. 
![image](https://user-images.githubusercontent.com/6158929/155835731-45a14433-56f7-47a1-909a-8c0b61aedc7a.png)

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?
Seems to be similar to [#5983](https://github.com/wagtail/wagtail/issues/5983)

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (yes / no)
Yes

### Technical details

* Python version: Run `python --version`. 3.6 -> 3.9
* Django version: Look in your requirements.txt, or run `pip show django | grep Version`. 2.2 -> 3.2
* Wagtail version: Look at the bottom of the Settings menu in the Wagtail admin, or run `pip show wagtail | grep Version:`. 2.13 -> 2.15.4
* Browser version: You can use https://www.whatsmybrowser.org/ to find this out. Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36

**Comment by thibaudcolas** [comment]
Thank you for the report @jslay-excella! It was so clear-cut that none of us replied 😅 

This should be fixed as of #9319, to be released in Wagtail 4.1.

**Comment by jslay-excella** [comment]
I had to re-read all of it to remember what it was even about :P 

Thanks!
